### PR TITLE
Add monitoring list and details pages for Alertmanager silences

### PR DESCRIPTION
**Pull request by kyoto**
Adds a list page and details page for Alertmanager Silences (rules that "silence" alerts).

`AlertsPage` and `SilencesPage` are very similar, so the common logic is extracted into `MonitoringListPage`.

Changes the sidebar link from `/monitoring/alerts` to `/monitoring` so that it remains highlighted on the new `/monitoring/silences*` pages.

Some things were left out to keep the PR smaller:
- List of Silenced Alerts, which will be added to the Silence details page in a future PR
- Labels list (because there is no Silences search page to link to yet)
- e2e tests

screenshot-1
screenshot-2

### CODE REVIEW

File: frontend/public/components/silence.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      </div>
+    </div>
+    <div className="col-xs-2">{numSilencedAlerts === undefined ? '-' : numSilencedAlerts}</div>
+  </ResourceRow>;
+});
+
+const SilenceHeader = props => <ListHeader>
+  <ColHead {...props} className="col-xs-7" sortField="name">Name</ColHead>
+  <ColHead {...props} className="col-xs-3" sortField="status.state">State</ColHead>
+  <ColHead {...props} className="col-xs-2" sortField="labels.severity">Silenced Alerts</ColHead>
```
Comment (spadgett) [inline frontend/public/components/silence.tsx]: This sort field doesn't look right

File: frontend/public/style/_icons.scss [hunk 2]
```diff
@@ -26,6 +27,7 @@
   color: $color-red-error;
 }
 
-.alert-pending {
+.alert-pending,
+.silence-pending {
```
Comment (spadgett) [inline frontend/public/style/_icons.scss]: Should this be the warning orange or text-muted gray? (I don't know alert manager well enough to say.)

Comment (kyoto) [reply frontend/public/style/_icons.scss]: The "expired" Silence icon is gray, so I think it makes sense to leave it as orange to more easily see the difference.

File: frontend/public/components/silence.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  expired: 'fa fa-times-circle-o text-muted',
+};
+
+const State = ({silence}) => {
+  const state = silenceState(silence);
+  if (state === 'inactive') {
+    return <span className="text-muted">{_.startCase(state)}</span>;
+  }
+  const klass = stateToIconClassName[state];
+  return klass ? <React.Fragment><i className={klass}></i> {_.startCase(state)}</React.Fragment> : null;
```
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/public/components/silence.tsx]: 👍

File: frontend/public/components/alert.tsx [hunk 4]
```diff
@@ -6,38 +6,24 @@ import { connect } from 'react-redux';
...
-  path: '/monitoring/alerts/rules',
-  abbr: 'AR',
-};
-
-const reduxID = 'monitoringRules';
-
 const detailsURL = (resource, name, labels) => `${resource.path}/${name}?${_.map(labels, (v, k) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`).join('&')}`;
 
-const ExternalLink = ({href, text}) => <a className="co-external-link" href={href} target="_blank" rel="noopener noreferrer">{text}</a>;
+export const ExternalLink = ({href, text}) => <a className="co-external-link" href={href} target="_blank" rel="noopener noreferrer">{text}</a>;
```
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/public/components/alert.tsx]: Moved to `utils/link.jsx`

File: frontend/public/components/silence.tsx [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+import { monitoringAlertsToProps, monitoringSilencesToProps } from '../ui/ui-reducers';
+import { connectMonitoringPage, ExternalLink, MonitoringListPage, MonitoringResourceIcon } from './alert';
+import { ColHead, ListHeader, ResourceRow } from './factory';
+import { SectionHeading, StatusBox, Timestamp } from './utils';
+
+const silenceLabels = silence => _.mapValues(_.keyBy(_.get(silence, 'matchers'), 'name'), 'value');
+
+const stateToIconClassName = {
+  active: 'fa fa-check-circle-o silence-active',
+  pending: 'fa fa-clock-o silence-pending',
```
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/public/components/silence.tsx]: I changed it to the hourglass.

File: frontend/public/components/silence.tsx (hunk 1 above)
Comment (kyoto) [reply frontend/public/components/silence.tsx]: Thanks. Actually, this sort is a bit tricky to implement, so I just removed the sort for now.

**Comment by kyoto** [comment]
@spadgett I've addressed your comments and rebased. Could you please take another look?

**Pull request merged**